src/C3 DFS BFS/P02 JS2 미로탈출 BFS.js

```diff
@@ -0,0 +1,44 @@
+function solution(arr) {
+  const n = arr.length;
+  const m = arr[0].length;
+  const dr = [-1, 1, 0, 0];
+  const dc = [0, 0, -1, 1];
+
+  function bfs(r, c) {
+    let que = [];
+    que.push([r, c]);
+
+    while (que.length !== 0) {
+      const row = que[0][0];
+      const col = que[0][1];
+      que.shift();
```

Comment:
```suggestion
      const [row, col] = que.shift();
```

`Array.prototype.shift`는 O(N)이기 때문에 N이 꽤 크면 시간초과가 될 수 있어요 :dog:

Reply:
이번 스터디를 계기로 shift 사용시 시간소모가 꽤 크다는 것을 알게되었네요 👶